### PR TITLE
code analysis fixes

### PR DESCRIPTION
**Pull request by akrantz**
Code analysis found cases where:
* parameters should be passed by reference.
* constants in header files should use constexpr to avoid multiple data copies.

### CODE REVIEW

**Comment by akrantz** [comment]
@kfarnung Please double-check that this builds for you.

File: lib/Debugger.Protocol/Generated/protocol/Protocol.h [hunk 1]
```diff
@@ -561,7 +561,7 @@ class MaybeBase {
     MaybeBase() : m_isJust(false) { }
     MaybeBase(T value) : m_isJust(true), m_value(value) { }
     MaybeBase(MaybeBase&& other) : m_isJust(other.m_isJust), m_value(std::move(other.m_value)) { }
-    void operator=(T value) { m_value = value; m_isJust = true; }
+    void operator=(const T& value) { m_value = value; m_isJust = true; }
```
[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.Protocol/Generated/protocol/Protocol.h]: Maybe, I definitely don't think this is the correct approach.

[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.Protocol/Generated/protocol/Protocol.h]: Can we revert this change for now, address the comments below and land this PR?  Then we can figure out how to deal with the dependencies in a separate PR.

[the remark this answers is not in the document]
Comment (akrantz) [reply lib/Debugger.Protocol/Generated/protocol/Protocol.h]: Will do.

File: lib/Debugger.ProtocolHandler/ProtocolHandler.cpp [hunk 2]
```diff
@@ -183,7 +183,7 @@ namespace JsDebug
         } while (m_waitingForDebugger || !current.empty());
     }
 
-    void ProtocolHandler::EnqueueCommand(ProtocolHandler::CommandType type, std::string message)
+    void ProtocolHandler::EnqueueCommand(ProtocolHandler::CommandType type, std::string const& message)
```
[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.ProtocolHandler/ProtocolHandler.cpp]: nit: `const std::string&` to be consistent with other locations

File: lib/Debugger.ProtocolHandler/ProtocolHandler.h [hunk 3]
```diff
@@ -56,7 +56,7 @@ namespace JsDebug
         };
 
         void SendResponse(const char* response);
-        void EnqueueCommand(CommandType type, std::string message = "");
+        void EnqueueCommand(CommandType type, std::string const& message = "");
```
[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.ProtocolHandler/ProtocolHandler.h]: nit: `const std::string&` to be consistent with other locations

[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.ProtocolHandler/ProtocolHandler.h]: Nevermind, page hadn't updated!

File: lib/Debugger.ProtocolHandler/ProtocolHandler.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (kfarnung) [reply lib/Debugger.ProtocolHandler/ProtocolHandler.cpp]: Nevermind, page hadn't updated!

**Pull request merged**